org.eclipse.winery.bpmn4tosca.converter.tobpel/src/test/java/org/eclipse/winery/bpmn2bpel/BPMN4Tosca2BpelTest.java

```diff
@@ -13,50 +13,135 @@
  *******************************************************************************/
 package org.eclipse.winery.bpmn2bpel;
 
-import java.net.MalformedURLException;
+import org.eclipse.winery.bpmn2bpel.parser.ParseException;
+import org.eclipse.winery.bpmn2bpel.planwriter.PlanWriterException;
+
+
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.nio.file.Paths;
+import java.io.File;
+import java.io.IOException;
 
-import org.eclipse.winery.bpmn2bpel.parser.ParseException;
-import org.eclipse.winery.bpmn2bpel.planwriter.PlanWriterException;
-
+import org.junit.rules.TemporaryFolder;
+import static org.junit.Assert.fail;
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
-
+import org.junit.Rule;
 
 public class BPMN4Tosca2BpelTest {
 
 	protected static String RESOURCES_DIR = "src/test/resources/bpmn4tosca";
 
+	/**
+	 *  Temporary Folder instance to keep out Zips from Sources.
+	 */
+	@Rule
+	private TemporaryFolder folder;
 
+	/**
+	 * Folder where Temporary Zip files will be created.
+	 */
+	private File temporaryFolder;
 	@Before
 	public void setUp() throws Exception {
-
+		folder = new TemporaryFolder();
+		folder.newFolder();
 	}
 
 	@After
 	public void tearDown() throws Exception {
 	}
 
 	@Test
-	public void testTransform() throws ParseException, PlanWriterException, MalformedURLException, URISyntaxException {
+	public void testTransform() throws ParseException,
+			PlanWriterException,
+			URISyntaxException, IOException{
 		URI srcUri = Paths.get(RESOURCES_DIR, "bpmn4tosca.json").toUri();
-		URI targetUri = Paths.get(RESOURCES_DIR, "managementplan.zip").toUri();
+		URI targetUri = Paths.get(RESOURCES_DIR, folder.newFile("managementplan.zip").getAbsolutePath()).toUri();
 		BPMN4Tosca2BpelTest.class.getResource(".");
 		Bpmn4Tosca2Bpel transformer = new Bpmn4Tosca2Bpel();
 		transformer.transform(srcUri, targetUri);
 	}
-	
+
 	@Test
 	public void testTransformGateway()
-			throws ParseException, PlanWriterException, MalformedURLException, URISyntaxException {
+			throws ParseException,
+			PlanWriterException,
+			URISyntaxException, IOException{
 		URI srcUri = Paths.get(RESOURCES_DIR, "bpmn4tosca.exclusivegateway.json").toUri();
-		URI targetUri = Paths.get(RESOURCES_DIR, "managementplan.exclusivegateway.zip").toUri();
+		URI targetUri = Paths.get(RESOURCES_DIR,   folder.newFile("managementplan.exclusivegateway.zip").getAbsolutePath()).toUri();
 		BPMN4Tosca2BpelTest.class.getResource(".");
 		Bpmn4Tosca2Bpel transformer = new Bpmn4Tosca2Bpel();
 		transformer.transform(srcUri, targetUri);
 	}
 
+
+	@Test
+	public void testTransformExceptionMissingNode() throws ParseException,
```

Comment:
Exceptions in JUnit are tested differently: See https://github.com/junit-team/junit4/wiki/Exception-testing

I would use the first option `@Test(expected = IndexOutOfBoundsException.class)`. In case you need something more sophisticated, use the other options described there.